Content/Items/Gear/Weapons/Sword/BloodOath.cs

```diff
@@ -0,0 +1,186 @@
+using PathOfTerraria.Core.Systems.Affixes.ItemTypes;
+using ReLogic.Content;
+using Terraria;
+using Terraria.DataStructures;
+using Terraria.ID;
+using Terraria.Localization;
+
+namespace PathOfTerraria.Content.Items.Gear.Weapons.Sword;
+
+internal class BloodOath : Sword
+{
+	public override string Texture => $"{PathOfTerraria.ModName}/Assets/Items/Gear/Weapons/Sword/BloodOath";
+	public override float DropChance => 5f;
+	public override int ItemLevel => 1;
+	public override bool IsUnique => true;
+	public override string Description => Language.GetTextValue("Mods.PathOfTerraria.Items.BloodOath.Description");
+	public override string AltUseDescription => Language.GetTextValue("Mods.PathOfTerraria.Items.BloodOath.AltUseDescription");
+	protected override bool CloneNewInstances => true;
+
+	private readonly HashSet<int> _hitNpcs = [];
+
+	private bool _specialOn = false;
+
+	public override ModItem Clone(Item newEntity)
+	{
+		// Items with special fields require cloning for things like dropping in world, splitting stacks (when applicable), displaying tooltips,
+		// and some swapping operations (such as moving from or to chests).
+		// This and CloneNewInstances makes sure the special isn't turned off or on when not desired.
+		ModItem clone = base.Clone(newEntity);
+		(clone as BloodOath)._specialOn = _specialOn;
+		return clone;
+	}
+
+	public override void Defaults()
+	{
+		base.Defaults();
+		Item.damage = 8;
+		Item.width = 58;
+		Item.height = 58;
+		Item.UseSound = SoundID.Item1;
+		Item.shoot = ProjectileID.None;
+	}
+	
+	public override string GenerateName()
+	{
+		return $"[c/FF0000:{Language.GetTextValue("Mods.PathOfTerraria.Items.BloodOath.DisplayName")}]";
+	}
+
+	public override List<ItemAffix> GenerateAffixes()
+	{
+		var sharpAffix = (ItemAffix)Affix.CreateAffix<AddedDamageAffix>(0, 2, 5);
+		var lifeAffix = (ItemAffix)Affix.CreateAffix<AddedLifeAffix>(0, 10, 10); // Add 10% life
+		return [sharpAffix, lifeAffix];
+	}
+	
+	public override bool AltFunctionUse(Player player)
+	{
+		AltUsePlayer modPlayer = player.GetModPlayer<AltUsePlayer>();
+
+		if (!modPlayer.AltFunctionAvailable)
+		{
+			return false;
+		}
+
+		_specialOn = !_specialOn;
+
+		if (!_specialOn)
+		{
+			foreach (int npcWho in _hitNpcs)
+			{
+				NPC npc = Main.npc[npcWho];
+
+				if (npc.active && npc.TryGetGlobalNPC(out BloodOathNPC bloodNpc))
+				{
+					bloodNpc.PopStacks(npc);
+				}
+			}
+
+			modPlayer.SetAltCooldown(1 * 60, 0);
+		}
+
+		return false;
+	}
+
+	public override void HoldItem(Player player)
+	{
+		if (_specialOn && Main.rand.NextBool(12))
+		{ 
+			Dust.NewDust(player.GetFrontHandPosition(Player.CompositeArmStretchAmount.None, 4f) + new Vector2(0, 4), 1, 1, DustID.Firework_Red);
+		}
+	}
+
+	public override void OnHitNPC(Player player, NPC target, NPC.HitInfo hit, int damageDone)
+	{
+		if (_specialOn)
+		{
+			var deathReason = PlayerDeathReason.ByCustomReason(Language.GetTextValue("Mods.PathOfTerraria.Items.BloodOath.DeathReason"));
```

Comment:
KEKW

Reply:
DeathReasons are necessary whenever damage is dealt. It's funny tho yeah - anything that deals damage has an associated death reason, like the Companion Cube's "pokes":
https://terraria.wiki.gg/wiki/Companion_Cube